### ISSUE TITLE
Improve CountryRule interface

### ISSUE THREAD

**Issue by otbutz** [graphhopper/graphhopper#2592]
Probably a good idea to gather the discussions from various issues and pull requests in a single ticket.

Previous discussions:

* https://github.com/graphhopper/graphhopper/pull/2450
* https://github.com/graphhopper/graphhopper/pull/2402
* https://github.com/graphhopper/graphhopper/pull/2268
* https://github.com/graphhopper/graphhopper/issues/2470

Open questions:
* how to deal with EncodedValue dependencies? (#2470)
* do we need the TransporationMode or should we replace it with dedicated methods? e.g `getMaxRegularSpeed()` / `getMaxHGVSpeed()`
* which level of detail do we want to express? e.g country specific bike dismount rules (#2402)
* do we even need maximum speed per country or should we rather obtain the average speed from the rules? (#2268)
* should we enable CountryRules by default?

**Comment by otbutz** [comment]
@easbar should we incorporate https://github.com/graphhopper/graphhopper/pull/2637 into the maxspeed resolution?

Looking at the [OSM wiki](https://wiki.openstreetmap.org/wiki/OSM_tags_for_routing/Maxspeed#Default_speed_limits_by_country) this would be enough to properly resolve speedlimits.

**Comment by easbar** [comment]
Yes, the urban_density and country encoded values should be used to determine the maxspeed. I think we just did not decide how to do this exactly yet. In this branch I tried calling the Kotlin lib that extracts the maxspeed from the json file that is created from the wiki table: https://github.com/graphhopper/graphhopper/compare/default_legal_speeds, but maybe we could also 'just' put this into the country rules.
